dag_in_context/src/schedule.rs

```diff
@@ -15,26 +14,33 @@ pub(crate) fn helpers() -> String {
         ; memory-helpers TODO run memory helpers for memory optimizations
     )
 
-    (saturate drop)
-    apply-drop-unions
-    cleanup-drop
+    
+    (saturate
+        (saturate
+          type-analysis
+          (saturate type-helpers))
+        error-checking
+        always-run
+        (saturate drop)
+        apply-drop-unions
+        cleanup-drop
 
-    (saturate subst)
-    apply-subst-unions
-    cleanup-subst
-)
+        (saturate subst)
+        apply-subst-unions
+        cleanup-subst)
```

Comment:
`subst` generates `AddContext` that needs to be resolved by the `context` ruleset

Reply:
Good catch!